hnn_core/network.py

```diff
@@ -253,7 +253,7 @@ def pick_connection(net, src_gids=None, target_gids=None,
         inner_set = set()
         # Union of indices which match inputs for single parameter
         for term in search_terms:
-            inner_set = inner_set.union(search_dict[term])
+            inner_set = inner_set.union(search_dict.get(term, list()))
```

Comment:
Controls for the rare case when `search_dict` is empty.

Reply:
Perfect for adding a unit test